packages/react-examples/.storybook/main.js

```diff
@@ -16,31 +18,33 @@ export default {
     // (also appears that it would require more configuration to work properly)
     reactDocgen: false,
   },
-  webpackFinal: (/** @type {import("webpack").Configuration} */ config) => {
+  webpackFinal: config => {
     const customConfig = custom(config);
 
-    customConfig.module?.rules?.push({
-      // Special loader that only includes stories from the current package
-      test: /\.storybook[/\\]preview.js/,
-      loader: path.resolve(__dirname, 'preview-loader.js'),
-    });
-
-    return {
-      ...customConfig,
+    return merge(customConfig, {
```

Comment:
to add some context:

I hit the wall hard with using the merge approach. lot of magic and unexpected behaviour can happen because of that. Also if one needs to use, it usually points that it is a code smell in overcomplicating configuration. 

I'll approve for now

Reply:
Hmm, that's too bad. If it was reliable, it actually seems much cleaner to me, rather than having to manually spread things and hope you didn't accidentally overwrite stuff. Or for example, if `config.module.rules` happened to be undefined, previously the new rule wouldn't have been added.

Reply:
Thanks for the approval! I merged this to get things off my plate but feel free to make further changes or revert this or later if you want.